enclave_manager/avalon_enclave_manager/singleton/config.toml

```diff
@@ -0,0 +1,115 @@
+# Set this flag to 1 for sync workorder execution.Else set it to 0
+sync_workload_execution = 1
+
+# -------------------------------------------------------------------
+# EnclaveManager -- configuration of Intel SGX Enclave EnclaveManager
+# -------------------------------------------------------------------
+[EnclaveManager]
+# configurations for asynchronous work load execution
+sleep_interval = "10"
+# configurations for synchronous Work load execution
+# zmq_port is the port used for zmq socket communication
+# Port for socket interaction with listener
+zmq_port = '5555'
+
+# -------------------------------------------------------
+# EnclaveModule -- configuration of the Intel SGX Enclave
+# -------------------------------------------------------
+[EnclaveModule]
+# Service Provider ID (SPID) is a 32-digit hex string tied to the
+# enclave implementation. Replace dummy SPID with value obtained after
+# subscription to run TCS in Intel SGX HW mode.
+spid = "DEADBEEF00000000DEADBEEF00000000"
+
+num_of_enclaves = "1"
+
+# ias_url is the URL of the Intel Attestation Service (IAS) server.
+ias_url = "https://api.trustedservices.intel.com/sgx/dev"
+
+# Proxy for https. Leave commented out for direct Internet connections or
+# uncomment and change to your corporate proxy.
+#https_proxy = "http://your-proxy:your-port/"
+
+# IAS API key is a 32-digit hex string subscription key used for authentication
+# of requests submitted to the IAS server. Obtain the key by subscribing in
+# the portal https://api.portal.trustedservices.intel.com/
+ias_api_key = "112233445566778899aabbccddeeff00"
+
+# TEE enclave library to use.
+enclave_library = "libtcf-enclave.signed.so"
+enclave_library_path = "tc/sgx/trusted_worker_manager/enclave/build/lib/"
+
+# -------------------------------------------------------------
+# TCS Worker configuration details
+# -------------------------------------------------------------
+
+[KMEListener]
+# URL of KMEListener where a Work order processor can send requests meant for
+# the Key Management Enclave(KME)  
+bind = "http://kme-listener:1948" 
+
+[WorkerConfig]
+ProofDataType = "TEE-SGX-IAS"
+
+# The Following URIs are used in direct model to submit work orders in
+# synchronous, asynchronous, pull and notification modes.
+WorkOrderSyncUri = "http://localhost:8080"
+WorkOrderAsyncUri = "http://localhost:8080"
+WorkOrderPullUri = "http://localhost:8080"
+WorkOrderNotifyUri = "http://localhost:8080"
+# Replace ApplicationTypeId and OrganizationId with real hexadecimal values
+ApplicationTypeId = "11aa22bb33cc44dd"
+OrganizationId = "aabbcc1234ddeeff"
+
+# URI used by worker to process work order receipts
+ReceiptInvocationUri = "http://localhost:8080"
+
+# Deployed address of Work order invocation smart contract used in proxy model
+WorkOrderInvocationAddress = ""
+# Work Order Receipt smart contract used in proxy model
+ReceiptInvocationAddress = ""
+# Ethereum Address used by the worker to submit transactions
+FromAddress = ""
```

Comment:
Why are these configurations kept in singleton enclave manager config. Singleton enclave manager is agnostic of proxy or direct model. These are meant for clients and block chain connectors. You could remove this from singleton and config and keep in top level tcs_config.toml. These are common data types for all workers (TC Spec section 8.1) and not specific to SGX TEE worker.

Reply:
In that case, we would land up using both the config files. It's manageable as long as building images/executable is done locally from codebase. 
Going ahead, when docker/executable are pulled from repositories, how do we foresee using mutiple config files?  Would duplication of few configs make more sense there?

Reply:
WorkOrderInvocationAddress , ReceiptInvocationAddress , FromAddress  should be generated as part of workerDetails. We are not using these parameters now. You can remove these params from all config files and assign it to empty in WorkerDetails.

Reply:
Updated @manojgop . Not expecting the configs now.